components/camera/videosource/webcam.go

```diff
@@ -202,7 +204,33 @@ func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logge
 	constraints := makeConstraints(attrs, debug, logger)
 
 	if attrs.Path != "" {
-		return tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		cam, err := tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		if err != nil {
+			return cam, err
+		}
+
+		goutils.PanicCapturingGo(func() {
+			src := camera.SourceFromCamera(cam)
```

Comment:
Very solid. We love a good fix! My only thing is maybe SourceFromCamera should return an error instead of just a nil src (when things go wrong). Then also, maybe check that error in here and handle it however if you can't get a source from ya camera.

Reply:
Good call. Done!